client/core/core.go

```diff
@@ -2013,85 +1918,68 @@ func (c *Core) connectDEX(acctInfo *db.AccountInfo) (*dexConnection, error) {
+	for i := range dexCfg.Assets {
+		asset := &dexCfg.Assets[i]
+		assets[asset.ID] = convertAssetInfo(asset)
+	}
+	// Validate the markets so we don't have to check every time later.
+	for _, mkt := range dexCfg.Markets {
+		_, ok := assets[mkt.Base]
+		if !ok {
+			log.Errorf("%s reported a market with base asset %d, "+
+				"but did not provide the asset info.", uri, mkt.Base)
 		}
-		// Validate the markets so we don't have to check every time later.
-		for _, mkt := range dexCfg.Markets {
-			_, ok := assets[mkt.Base]
-			if !ok {
-				log.Errorf("%s reported a market with base asset %d, "+
-					"but did not provide the asset info.", uri, mkt.Base)
-			}
-			_, ok = assets[mkt.Quote]
-			if !ok {
-				log.Errorf("%s reported a market with quote asset %d, "+
-					"but did not provide the asset info.", uri, mkt.Quote)
-			}
+		_, ok = assets[mkt.Quote]
+		if !ok {
+			log.Errorf("%s reported a market with quote asset %d, "+
+				"but did not provide the asset info.", uri, mkt.Quote)
 		}
+	}
 
-		marketMap := make(map[string]*Market)
-		for _, mkt := range dexCfg.Markets {
-			base, quote := assets[mkt.Base], assets[mkt.Quote]
-			market := &Market{
-				Name:            mkt.Name,
-				BaseID:          base.ID,
-				BaseSymbol:      base.Symbol,
-				QuoteID:         quote.ID,
-				QuoteSymbol:     quote.Symbol,
-				EpochLen:        mkt.EpochLen,
-				StartEpoch:      mkt.StartEpoch,
-				MarketBuyBuffer: mkt.MarketBuyBuffer,
-			}
-			marketMap[mkt.Name] = market
+	marketMap := make(map[string]*Market)
+	for _, mkt := range dexCfg.Markets {
+		base, quote := assets[mkt.Base], assets[mkt.Quote]
+		market := &Market{
+			Name:            mkt.Name,
+			BaseID:          base.ID,
+			BaseSymbol:      base.Symbol,
+			QuoteID:         quote.ID,
+			QuoteSymbol:     quote.Symbol,
+			EpochLen:        mkt.EpochLen,
+			StartEpoch:      mkt.StartEpoch,
+			MarketBuyBuffer: mkt.MarketBuyBuffer,
 		}
+		marketMap[mkt.Name] = market
+	}
 
-		// Create the dexConnection and add it to the map.
-		dc := &dexConnection{
-			WsConn:     conn,
-			connMaster: connMaster,
-			assets:     assets,
-			cfg:        dexCfg,
-			books:      make(map[string]*bookie),
-			acct:       newDEXAccount(acctInfo),
-			marketMap:  marketMap,
-			trades:     make(map[order.OrderID]*trackedTrade),
-			notify:     c.notify,
-		}
+	// Create the dexConnection and listen for incoming messages.
+	dc := &dexConnection{
+		WsConn:     conn,
+		connMaster: connMaster,
+		assets:     assets,
+		cfg:        dexCfg,
+		books:      make(map[string]*bookie),
+		acct:       newDEXAccount(acctInfo),
+		marketMap:  marketMap,
+		trades:     make(map[order.OrderID]*trackedTrade),
+		notify:     c.notify,
+	}
 
-		dc.refreshMarkets()
-		connChan <- dc
-		c.wg.Add(1)
-		// Listen for incoming messages.
-		log.Infof("Connected to DEX %s and listening for messages.", uri)
-		go c.listen(dc)
-	}) // End 'config' request
+	dc.refreshMarkets()
+	go c.listen(dc)
```

Comment:
I ended up refactoring this dex connection listener in https://github.com/decred/dcrdex/pull/328/commits/64ffbd5dd243c7a06dfedc3e09775efb7833a6fc.

My reasons:
- It reads better having the dex connection perform the listening -> dc.listen(c) rather than c.listen(dc).
- Moved the wg incrementing to the listen method but outside the actual listen goroutine because keeping calls to `wg.Add()` and `wg.Done()` close makes it easier to understand how the wg is used / why it is incremented.

Please let me know if that's out of place / potentially problematic so I can revert. I put that change set in a separate commit so I can revert easily if necessary.

Reply:
This is now the only place in **core** where the `dexConnection` handles a reference to the `Core` instead of the other way around. It breaks a larger scheme of separation of concerns. It's also out of scope for this PR, and comments like "It reads better having the dex connection perform the listening" are subjective.

In general, you shouldn't be making any changes to type names, variable names, or general code structure outside of the scope of the pull request, and usually not without a prior discussion or issue.

Reply:
Noted. Will undo. I sometimes get lost in code and overdo. Apologies.

Reply:
> I sometimes get lost in code and overdo. Apologies.

No worries. I'm often guilty of that myself.